### PR TITLE
Stopping CHES Call when no email provided on Submit

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Stopping the CHES call when there are no emails provided.

## Types of changes
Made changes to the emailService so that it checks the contexts variable for any emails to send to.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/forms/email/emailService.js [hunk 1]
```diff
@@ -287,8 +287,9 @@ const service = {
   submissionReceived: async (formId, submissionId, body, referer) => {
     try {
       const { configData, contexts } = await buildEmailTemplate(formId, submissionId, EmailTypes.SUBMISSION_RECEIVED, referer, { body });
-
-      return service._sendEmailTemplate(configData, contexts);
+      if (contexts[0].to.length > 0) {
```
Comment (jujaga) [inline app/src/forms/email/emailService.js:290]: Use shorthand truthiness check:
```js
if (contexts[0].to.length) {
```

File: app/src/forms/email/emailService.js [hunk 2]
```diff
@@ -287,8 +287,9 @@ const service = {
...
+        return service._sendEmailTemplate(configData, contexts);
```
Comment (jujaga) [inline app/src/forms/email/emailService.js:291]: The specification of this function needs to return the result of the operation. By if-blocking this out, this function now returns nothing; if the caller function depends on the returned object, we're introducing a potential bug. We should have an else return { ...something}.

**Review by jujaga — CHANGES_REQUESTED** [review]

File: app/src/forms/email/emailService.js (hunk 1 above)
Comment (loneil) [reply app/src/forms/email/emailService.js:290]: I'm not sure this should be based on checking contexts in the email service. Aren't the contexts set by the supplied email address in the body, so shouldn't it check a level or 2 higher to even see if email service should be called at all?

So check body email address at the controller level. The code flow is really "do I need to do anything about an email here?" when controlling the submission POST

Comment (loneil) [reply app/src/forms/email/emailService.js:290]: IE, controller.createSubmission line 126

File: app/src/forms/email/emailService.js [hunk 3]
```diff
@@ -287,8 +287,11 @@ const service = {
   submissionReceived: async (formId, submissionId, body, referer) => {
     try {
       const { configData, contexts } = await buildEmailTemplate(formId, submissionId, EmailTypes.SUBMISSION_RECEIVED, referer, { body });
-
-      return service._sendEmailTemplate(configData, contexts);
+      if (contexts[0].to.length) {
+        return service._sendEmailTemplate(configData, contexts);
+      } else {
+        return {};
+      }
```
Comment (loneil) [inline app/src/forms/email/emailService.js:294]: Really don't think this is the right place to be doing this? It's not the email service deciding based on the CHES contexts whether to send, it should be the CONTROLLER deciding to invoke email service based on the request
See other comment below:

I'm not sure this should be based on checking contexts in the email service. Aren't the contexts set by the supplied email address in the body, so shouldn't it check a level or 2 higher to even see if email service should be called at all?

So check body email address at the controller level. The code flow is really "do I need to do anything about an email here?" when controlling the submission POST

IE, controller.createSubmission line 126

**Review by loneil — COMMENTED** [review]
@jujaga your thoughts?

**Comment by loneil** [comment]
Actually checking it out the form detail is only known at the `buildEmailTemplate` point so my other suggestion might not be valid... though still a little uneasy about only doing this contexts check in one of the email handlers when it really would apply to all... but it should fix this particular issue at least I think. Might consider refactoring later to apply more globally?

**Review by loneil — APPROVED** [review]
This should work, discussed about considering that "is there a `to` address" check in other stuff but can consider later

**Review by TimCsaky — APPROVED** [review]

**Review by jujaga — CHANGES_REQUESTED** [review]
Please do a proper rebase of this branch.

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-280

**Review by jujaga — APPROVED** [review]

**Pull request merged**